Fix flaky E2E test that prematurely kills entire test job

Error seen here - https://travis-ci.org/ampproject/amphtml/jobs/531973468#L1559

```
[20:36:16] Starting 'e2e'...
●[20:36:45] 'e2e' errored after 29 s
[20:36:45] AssertionError: expected 'https://players.brightcove.net/1290862519001/SyIOV8yWM_default/index.html?videoId=ref:amp-docs-sample&playsinline=true' to include 'bound'
The command "node build-system/pr-check/e2e-tests.js" exited with 1.
```

--- a/build-system/tasks/e2e/describes-e2e.js
+++ b/build-system/tasks/e2e/describes-e2e.js
@@ -274,7 +274,7 @@ function describeEnv(factory) {
         asyncErrorTimerId = setTimeout(() => {
           const lastExpectError = getLastExpectError();
           if (lastExpectError) {
-            throw lastExpectError;
+            this.test.error(lastExpectError);
           }
         }, this.timeout() - 1);
         fn.call(this, env);

--- a/build-system/tasks/e2e/selenium-webdriver-controller.js
+++ b/build-system/tasks/e2e/selenium-webdriver-controller.js
@@ -325,6 +325,8 @@ class SeleniumWebDriverController {
   }
 
   /**
+   * Gets the element property. Note that this is different
+   * than getElementAttribute()
    * @param {!ElementHandle<!WebElement>} handle
    * @param {string} property
    * @return {!Promise<string>}

